packages/perseus/src/widgets/radio/radio-validator.ts

```diff
@@ -21,23 +23,33 @@ function radioValidator(
         };
     }
 
-    // TODO(LEMS-2541): should numCorrect actually be on the rubric
-    // instead of the userInput?
-    if (
-        userInput.numCorrect &&
-        userInput.numCorrect > 1 &&
-        numSelected !== userInput.numCorrect
-    ) {
+    const numCorrect: number = _.reduce(
+        rubric.choices,
+        function (memo, choice) {
+            return choice.correct ? memo + 1 : memo;
+        },
+        0,
+    );
+
+    if (numCorrect && numCorrect > 1 && numSelected !== numCorrect) {
         return {
             type: "invalid",
             message: strings.chooseCorrectNum,
         };
         // If NOTA and some other answer are checked, ...
     }
+    let noneOfTheAboveSelected = false;
+    for (let i = 0; i < rubric.choices.length; i++) {
+        if (
+            rubric.choices[i].isNoneOfTheAbove &&
+            userInput.choicesSelected[i]
+        ) {
+            noneOfTheAboveSelected = true;
+            break;
+        }
+    }
```

Comment:
Your code is clearer, but just because I like the new methods here is an alternative:

```suggestion
    let noneOfTheAboveSelected = rubric.choices.some((e, i) => (
         e.isNoneOfTheAbove && userInput.choicesSelected[i]
    )
```